src/widgets/scrollbar.rs

```diff
@@ -0,0 +1,208 @@
+use crate::{
+    buffer::Buffer,
+    layout::Rect,
+    style::Style,
+    symbols::{block::FULL, line::DOUBLE_VERTICAL},
+    widgets::Widget,
+};
+
+/// Scrollbar widget for tui-rs library.
+///
+/// This widget can be used to display a scrollbar in a terminal user interface.
+#[derive(Default, Debug, Clone)]
+pub struct Scrollbar<'a> {
+    scroll: f64,
+    style: Style,
+    symbol: &'a str,
+    bar_style: Style,
+    bar_symbol: &'a str,
+}
+
+impl<'a> Scrollbar<'a> {
+    /// Creates a new scrollbar with default styles and symbols.
+    pub fn new(scroll: f64) -> Self {
+        Self {
+            scroll,
+            style: Style::default(),
+            symbol: FULL,
+            bar_style: Style::default(),
+            bar_symbol: DOUBLE_VERTICAL,
+        }
+    }
+
+    /// Sets the scroll position of the scrollbar.
+    pub fn scroll(mut self, scroll: f64) -> Self {
+        if scroll < 0.0 {
+            self.scroll = 0.0;
+        } else {
+            self.scroll = scroll;
+        }
+        self
+    }
+
+    /// Sets the symbol that represents the scroller.
+    pub fn symbol(mut self, symbol: &'a str) -> Self {
```

Comment:
I wonder if we should just use https://crates.io/crates/derive_setters for new widgets.

Reply:
I'm assuming this can be done as a separate PR for all widgets, so I'll resolve this conversation for now.

Reply:
yep